src/crypto/SymmetricCipher.cpp

```diff
@@ -89,17 +91,58 @@ SymmetricCipher::Algorithm SymmetricCipher::cipherToAlgorithm(Uuid cipher)
     if (cipher == KeePass2::CIPHER_AES) {
         return SymmetricCipher::Aes256;
     }
-    else {
+    else if (cipher == KeePass2::CIPHER_CHACHA20) {
+        return SymmetricCipher::ChaCha20;
+    }
+    else if (cipher == KeePass2::CIPHER_TWOFISH) {
         return SymmetricCipher::Twofish;
     }
+
+    qWarning("SymmetricCipher::cipherToAlgorithm: invalid Uuid %s", cipher.toByteArray().toHex().data());
+    return static_cast<SymmetricCipher::Algorithm>(-1);
```

Comment:
Never return an invalid enum value. If you need to have some sort of error value, create a dedicated enum entry.